packages/gestalt/src/SheetMobile/FullPage.js

```diff
@@ -0,0 +1,270 @@
+                          accessibilityLabel={accessibilityDismissButtonLabel}
+                          accessibilityControls={id}
+                          onClick={() => onDismiss()}
+                          ref={dismissButtonRef}
+                          size="lg"
+                        />
+                      </Flex.Item>
+                    ) : null}
+                    {Boolean(heading) && (
+                      <Flex.Item flex="grow">
+                        <Flex direction="column">
+                          {typeof heading === 'string' ? (
+                            <Heading align={align} size="300" accessibilityLevel={1} lineClamp={2}>
+                              {heading}
+                            </Heading>
+                          ) : (
+                            heading
+                          )}
+                          {subHeading && (
+                            <Text align={align} size="100">
+                              {subHeading}
+                            </Text>
+                          )}
+                        </Flex>
+                      </Flex.Item>
+                    )}
+                    {forwardIconButton && !primaryAction ? (
+                      <Flex.Item flex="none">
+                        <IconButton
+                          accessibilityLabel={forwardIconButton.accessibilityLabel}
+                          icon="arrow-forward"
+                          iconColor="darkGray"
+                          size="lg"
+                          onClick={({ event }) =>
+                            forwardIconButton?.onClick({ event, onDismissStart: onExternalDismiss })
+                          }
+                        />
+                      </Flex.Item>
+                    ) : null}
+                    {primaryAction ? (
+                      // Allow button text to wrap on mobile
+                      <Flex.Item flex="shrink">
+                        <PrimaryAction
+                          accessibilityLabel={primaryAction.accessibilityLabel}
+                          href={primaryAction.href}
+                          rel={primaryAction?.rel}
+                          target={primaryAction?.target}
+                          label={primaryAction.label}
+                          onClick={({ event }) =>
+                            primaryAction?.onClick({ event, onDismissStart: onExternalDismiss })
+                          }
+                        />
+                      </Flex.Item>
+                    ) : null}
+                  </Flex>
+                </Box>
+
+                <Box
+                  padding={padding === 'none' ? 0 : 4}
+                  flex="grow"
+                  overflow="auto"
+                  onScroll={updateShadows}
+                  ref={contentRef}
+                >
+                  {children}
+                </Box>
+
+                {Boolean(footer) && (
```

Comment:
Let's say you have a keyboard input, and the footer says "Set Passcode". Will it appear over the keyboard and be pushed up, or will the keyboard cover the footer. Will the footer always appear on top of the content? If so, should content behind it be dimmed?